cmd/govim/buffer_events.go

```diff
@@ -61,7 +61,7 @@ func (v *vimstate) bufReadPost(args ...json.RawMessage) error {
 		v.Logf("failed to update signs for buffer %d: %v", nb.Num, err)
 	}
 
-	if err := v.redefineHighlights(v.diagnostics()); err != nil {
+	if err := v.redefineHighlights(v.diagnostics(), true); err != nil {
```

Comment:
I'm not sure this is guaranteed to be safe. Because it's possible a cheeky file change snuck in before we opened the file which would make the current diagnostics for that file "bad". Calling with `true` is correct, but per our Slack chat this is one situation where I think we could end up applying bad highlights.

Reply:
Yeah, we'll handle that in https://github.com/govim/govim/issues/685.